### PR TITLE
Fixed issue #18517: Survey editing : Survey creation date is wrong

### PR DESCRIPTION
**Pull request by gabrieljenik**

### CODE REVIEW

File: application/models/Survey.php [hunk 1]
```diff
@@ -550,7 +550,7 @@ public function rules()
             array('running', 'safe', 'on' => 'search'),
             array('expires', 'date','format' => ['yyyy-M-d H:m:s.???','yyyy-M-d H:m:s','yyyy-M-d H:m'],'allowEmpty' => true),
             array('startdate', 'date','format' => ['yyyy-M-d H:m:s.???','yyyy-M-d H:m:s','yyyy-M-d H:m'],'allowEmpty' => true),
```
Comment (Shnoulle) [inline application/models/Survey.php:552]: I know you want to fix each issue separately, but here seems there are same issue just 2 lines before.

Can we fix it at same time, it take less time to fix directly it than to report an issue and create another PR …

**Review by Shnoulle — COMMENTED** [review]
See comment

File: application/models/Survey.php (hunk 1 above)
Comment (Shnoulle) [reply application/models/Survey.php:552]: I can test i think , but yes : not really needed.

Comment (gabrieljenik) [reply application/models/Survey.php:552]: I would make a different PR, maybe a direct commit?

Comment (Shnoulle) [reply application/models/Survey.php:552]: Unsure  …

Comment (gabrieljenik) [reply application/models/Survey.php]: Undid the change. That format validator pattern was ok

Comment (Shnoulle) [reply application/models/Survey.php]: Then array('datecreated', 'default', 'value' => date("Y-m-d H:i:s")), didn't fix issue

File: application/models/Survey.php [hunk 2]
```diff
@@ -493,7 +493,7 @@ public function rules()
             array('sid', 'numerical', 'integerOnly' => true,'min' => 1), // max ?
             array('sid', 'unique'),// Not in pk
             array('gsid', 'numerical', 'integerOnly' => true),
-            array('datecreated', 'default', 'value' => date("Y-m-d H:m:s")),
+            array('datecreated', 'default', 'value' => date("Y-m-d H:i:s")),
```
[the remark this answers is not in the document]
Comment (Shnoulle) [reply application/models/Survey.php:496]: You give me the issue ;)

[the remark this answers is not in the document]
Comment (gabrieljenik) [reply application/models/Survey.php:496]: Well, that date default value is set by the date function, which uses "i" for minutes

[the remark this answers is not in the document]
Comment (Shnoulle) [reply application/models/Survey.php:496]: OK :)

**Review by Shnoulle — APPROVED** [review]

**Comment by Shnoulle** [comment]
Code review done

File: application/models/Survey.php (hunk 1 above)
Comment (gabrieljenik) [reply application/models/Survey.php]: This was hard to test. The issue is not on the copy process, but on the survey creation one, where the default value is given to the datecreated field

File: application/models/Survey.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (gabrieljenik) [reply application/models/Survey.php:496]: > You give me the issue ;)

That issue was for the format pattern, which is run by the Yii framework. That's where the confusion lies.

File: application/models/Survey.php (hunk 1 above)
Comment (Shnoulle) [reply application/models/Survey.php]: Yes, thank you for explanation

**Pull request merged**